components/builder-api/src/server/resources/jobs.rs

```diff
@@ -122,14 +125,56 @@ fn get_rdeps(req: HttpRequest,
     match route_message::<jobsrv::JobGraphPackageReverseDependenciesGet,
                         jobsrv::JobGraphPackageReverseDependencies>(&req, &rdeps_get)
     {
-        Ok(rdeps) => HttpResponse::Ok().json(rdeps),
+        Ok(rdeps) => {
+            let filtered = match filtered_rdeps(&req, rdeps) {
+                Ok(f) => f,
+                Err(err) => return err.into(),
+            };
+            HttpResponse::Ok().json(filtered)
+        }
         Err(err) => {
             debug!("{}", err);
             err.into()
         }
     }
 }
 
+fn filtered_rdeps(req: &HttpRequest,
+                  rdeps: jobsrv::JobGraphPackageReverseDependencies)
+                  -> Result<jobsrv::JobGraphPackageReverseDependencies> {
+    let mut new_rdeps = jobsrv::JobGraphPackageReverseDependencies::new();
+    new_rdeps.set_origin(rdeps.get_origin().to_string());
+    new_rdeps.set_name(rdeps.get_name().to_string());
+
+    let mut origin_map = HashMap::new();
+    let mut short_deps = RepeatedField::new();
+
+    for rdep in rdeps.get_rdeps() {
+        let ident = OriginPackageIdent::from_str(rdep)?;
+        let origin_name = ident.get_origin();
+        let pv = if !origin_map.contains_key(origin_name) {
+            let conn = req_state(req).db.get_conn().map_err(Error::DbError)?;
+            let origin = Origin::get(&origin_name, &*conn)?;
+            origin_map.insert(origin_name.to_owned(),
+                              origin.default_package_visibility.clone());
+            origin.default_package_visibility
+        } else {
+            origin_map.get(origin_name).unwrap().clone()
+        };
+        if pv != PackageVisibility::Public {
+            if authorize_session(req, Some(&origin_name)).is_err() {
+                debug!("Skipping unauthorized non-public origin package: {}", rdep);
+                continue; // Skip any unauthorized origin packages
+            }
+        }
+
+        short_deps.push(rdep.to_string())
+    }
+
+    new_rdeps.set_rdeps(short_deps);
+    return Ok(new_rdeps);
```

Comment:
Minor: return is unneeded here since it's the last line of the function. It can simply be: `Ok(new_rdeps)` with no semicolon.

Reply:
Good catch .. thanks